### ISSUE TITLE
HPA not working properly when pod status "Unknown" (node failure)

### ISSUE THREAD

**Issue by moonek** [kubernetes/kubernetes#62845]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
HPA not working properly when pod status `Unknown` (node failure)
I confirmed in k8s `v1.10.3` that this problem has not been solved yet.

**What you expected to happen**:
HPA based on number of normal pods except `Unknown` pods.

**How to reproduce it (as minimally and precisely as possible)**:
Test yaml
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: test-nginx-autoscaling
spec:
  replicas: 2
  template:
    metadata:
      labels:
        name: test-nginx-autoscaling
    spec:
      containers:
      - name: test-nginx-autoscaling
        image: myrepo/test-nginx:stress
        ports:
        - containerPort: 80
        resources:
          requests:
            cpu: 500m
            memory: 128Mi
---
apiVersion: autoscaling/v1
kind: HorizontalPodAutoscaler
metadata:
  name: test-nginx-autoscaling
  labels:
    name: test-nginx-autoscaling
spec:
  scaleTargetRef:
    apiVersion: apps/v1beta1
    kind: Deployment
    name: test-nginx-autoscaling
  minReplicas: 2
  maxReplicas: 4
  targetCPUUtilizationPercentage: 80
```
Initial state
```
NAME                                          READY     STATUS    RESTARTS   AGE       IP              NODE
po/test-nginx-autoscaling-6dcf47755f-kd64n    1/1       Running   0          1m       10.244.144.43   worker-3
po/test-nginx-autoscaling-6dcf47755f-x9vk4    1/1       Running   0          1m       10.244.56.17    worker-1
```

Intentionally caused a `worker-1` failure
```
NAME       STATUS     ROLES     AGE       VERSION
master-1   Ready      master    23d      v1.10.3
master-2   Ready      master    23d      v1.10.3
master-3   Ready      master    23d      v1.10.3
worker-1   NotReady   <none>    23d      v1.10.3
worker-2   Ready      <none>    23d      v1.10.3
worker-3   Ready      <none>    23d      v1.10.3
worker-4   Ready      <none>    23d      v1.10.3
worker-5   Ready      <none>    23d      v1.10.3
```

The pod was created on the other node, and the `worker-1` pod was `Unknown`
```
NAME                                          READY     STATUS    RESTARTS   AGE       IP              NODE
po/test-nginx-autoscaling-6dcf47755f-g8n55    1/1       Running   0          21m       10.244.176.24   worker-2
po/test-nginx-autoscaling-6dcf47755f-kd64n    1/1       Running   0          14m       10.244.144.43   worker-3
po/test-nginx-autoscaling-6dcf47755f-x9vk4    1/1       Unknown   0          21m       10.244.56.17    worker-1
```

Intentionally caused a CPU load on a particular pod.
```
[root@jumphost:/root] kubectl exec -it test-nginx-autoscaling-6dcf47755f-g8n55 bash
root@test-nginx-autoscaling-6dcf47755f-g8n55:/# stress -c 1
stress: info: [14] dispatching hogs: 1 cpu, 0 io, 0 vm, 0 hdd
```

The load continues, but there are not `3` `running` pods.
The `TARGETS` is `100%/80%`.

```
[root@jumphost:/root] kubectl get hpa,deploy,rs,po -owide

NAME                          REFERENCE                            TARGETS      MINPODS   MAXPODS   REPLICAS   AGE
hpa/test-nginx-autoscaling    Deployment/test-nginx-autoscaling    100% / 80%   2         4         2          1h

NAME                             DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE       CONTAINERS
IMAGES                                    SELECTOR
deploy/test-nginx-autoscaling    2         2         2            2           1h        test-nginx-autoscaling
myrepo/test-nginx:stress   name=test-nginx-autoscaling


NAME                                    DESIRED   CURRENT   READY     AGE       CONTAINERS                IMAGES
                                  SELECTOR
rs/test-nginx-autoscaling-6dcf47755f    2         2         2         1h        test-nginx-autoscaling    myrepo/test-nginx:stress   name=test-nginx-autoscaling,pod-template-hash=2879033119


NAME                                          READY     STATUS    RESTARTS   AGE       IP              NODE
po/test-nginx-autoscaling-6dcf47755f-g8n55    1/1       Running   0          1h        10.244.176.24   worker-2
po/test-nginx-autoscaling-6dcf47755f-kd64n    1/1       Running   0          1h        10.244.144.43   worker-3
po/test-nginx-autoscaling-6dcf47755f-x9vk4    1/1       Unknown   0          1h        10.244.56.17    worker-1
```

The number of `AVAILABLE` pods does not include the `Unknown` pod, the `HPA` logic seems to recognize the `Unknown` pod as a normal pod.
In addition, the `DESIRED` of deployments does not increase to `3`.

**Anything else we need to know?**:
If `Unknown` pod is not generated under the same condition, it is scaled normally.
```
po/test-nginx2-autoscaling-775668f75f-6449z   1/1       Running   0          1m        10.244.176.26   worker-2
po/test-nginx2-autoscaling-775668f75f-sb48n   1/1       Running   0          1m        10.244.144.8    worker-3
```
```
[root@jumphost:/root kubectl exec -it test-nginx2-autoscaling-775668f75f-6449z bash
root@test-nginx2-autoscaling-775668f75f-6449z:/# stress -c 1
stress: info: [20] dispatching hogs: 1 cpu, 0 io, 0 vm, 0 hdd
```
```
[root@jumphost:/root] kubectl get hpa,deploy,rs,po -owide
NAME                          REFERENCE                            TARGETS     MINPODS   MAXPODS   REPLICAS   AGE
hpa/test-nginx2-autoscaling   Deployment/test-nginx2-autoscaling   66% / 80%   2         4         3          10m

NAME                             DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE       CONTAINERS
IMAGES                                    SELECTOR
deploy/test-nginx2-autoscaling   3         3         3            3           10m        test-nginx2-autoscaling
myrepo/test-nginx:stress   name=test-nginx2-autoscaling

NAME                                    DESIRED   CURRENT   READY     AGE       CONTAINERS                IMAGES
                                  SELECTOR
rs/test-nginx2-autoscaling-775668f75f   3         3         3         10m        test-nginx2-autoscaling   myrepo/test-nginx:stress   name=test-nginx2-autoscaling,pod-template-hash=3312249319

NAME                                          READY     STATUS    RESTARTS   AGE       IP              NODE
po/test-nginx2-autoscaling-775668f75f-6449z   1/1       Running   0          10m        10.244.176.26   worker-2
po/test-nginx2-autoscaling-775668f75f-nb10mh   1/1       Running   0          4m        10.244.80.48    worker-4
po/test-nginx2-autoscaling-775668f75f-sb48n   1/1       Running   0          10m        10.244.144.8    worker-3
```

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:17:39Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:05:37Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
on-premise
- OS (e.g. from /etc/os-release):
```
CentOS Linux release 7.3.1611 (Core)
```
- Kernel (e.g. `uname -a`):
```
3.10.0-514.26.2.el7.x86_64
```
- Install tools:
kubeadm
- Others:
docker 1.12.6

**Comment by moonek** [comment]
I confirmed that the bug was fixed normally in v1.10.0. https://github.com/kubernetes/kubernetes/pull/60648
Close the issue.

**Comment by moonek** [comment]
I confirmed in k8s v1.10.3 that this problem has not been solved yet.
#60648 was not relevant.
